File: topicos/poo/java/ex1/main.java
public class main {
  public static void main(String[] args) {
    Livro livro_1 = new Livro("1234", "O senhor dos Aneis", "JRR Tolkien");
    Livro livro_2 = new Livro("4231", "O nome do Vento", "Patrick Ruthfuss");

    Leitor leitor_1 = new Leitor(29, "Joao Henrique Serodio", livro_1);
    Leitor leitor_2 = new Leitor(48, "Adriano Curuja", livro_2);

    livro_1.putLeitor(leitor_1);
    livro_2.putLeitor(leitor_2);
    livro_2.putLeitor(leitor_1);

    leitor_1.oqueEstalendo();
    leitor_2.oqueEstalendo();

    livro_1.printLeitores();
    livro_2.printLeitores();
  }
}

File: topicos/poo/node/ex1/main.js
import Leitor from './Leitor.js';
import Livro from './Livro.js';


function main() {
  // livros 
  const livro_1 = new Livro("1234", "O senhor dos Aneis", "JRR Tolkien");
  const livro_2 = new Livro("2342d", "A arte de ler e meditar", "Hugo de Sao Vitor");

  // leitores
  const leitor_1 = new Leitor(29, "Nereu de Oliveira", livro_1);
  const leitor_2 = new Leitor(36, "Joao Henrique Serodio", livro_2);

  // livro 2 tambem foi lido por leitor_1
  livro_2.incluirLeitor(leitor_1);

  leitor_1.oqueEstaLendo();
  leitor_2.oqueEstaLendo();

  // lista de livros
  livro_1.imprimirLeitores();
  livro_2.imprimirLeitores();
}

main();
